src/components/withViewportOffsetTop.js

```diff
@@ -13,43 +13,33 @@ const viewportOffsetTopPropTypes = {
 };
 
 export default function (WrappedComponent) {
-    class WithViewportOffsetTop extends Component {
-        constructor(props) {
-            super(props);
-
-            this.updateDimensions = this.updateDimensions.bind(this);
-
-            this.state = {
-                viewportOffsetTop: 0,
-            };
-        }
-
-        componentDidMount() {
-            this.removeViewportResizeListener = addViewportResizeListener(this.updateDimensions);
-        }
-
-        componentWillUnmount() {
-            this.removeViewportResizeListener();
-        }
+    function WithViewportOffsetTop(props) {
+        const [viewportOffsetTop, setViewportOffsetTop] = useState(0);
 
         /**
          * @param {SyntheticEvent} e
          */
-        updateDimensions(e) {
-            const viewportOffsetTop = lodashGet(e, 'target.offsetTop', 0);
-            this.setState({viewportOffsetTop});
-        }
+        const updateDimensions = useCallback((e) => {
+            const viewPortOffsetTop = lodashGet(e, 'target.offsetTop', 0);
+            setViewportOffsetTop(viewPortOffsetTop);
+        }, []);
```

Comment:
Seems like this function is only used for the listener below. We can move it inside the `useEffect`, then we have no need to wrap it in `useCallback`, and the effect dependencies will also be cleared